File: Source/CookItNow.Web/Scripts/app/models/quick-recipe.ts
import {Quantity} from "models/quantity"; 

export interface QuickRecipe {
    id : number;
    title : string;
    originalUrl : string;
    summary : string;
    originalServings : number;
    duration : Duration[];
    subrecipes : SubRecipe[];
    ingredients : Ingredient[];
    requirements: Requirement[];
    steps: Step[];
}

export interface Duration {
    title: string;
    time: string;
}

export interface SubRecipe {
    id: number;
    title: string;
}

export interface Ingredient {
    id: number;
    subrecipeId: number;
    name: string;
    quantity: Quantity;
    requirements: string[];
    replacements: Ingredient[];
}

export interface Requirement {
    action: string;
    ingredientId: number;
}

export interface Step {
    subrecipeId: number;
    phrases: Phrase[];
    postStep: Step;
}

export interface Phrase {
    parts: Part[];
}

export interface Part {
    type: string;
}

export interface IngredientPart extends Part {
    ingredient: Ingredient;
}

export interface TextPart extends Part {
    value: string;
}

export interface ActionPart extends Part {
    value: string;
}

export interface TimerPart extends Part {
    value: string;
    action: string;
}

File: Source/core2/E133.Business/Models/QuickRecipe.cs
using System.Collections.Generic;

namespace E133.Business.Models
{
    public class QuickRecipe : Document
	{
        public QuickRecipe()
        {
            this.Durations = new List<Duration>();
            this.Subrecipes = new List<Subrecipe>();
            this.Ingredients = new List<Ingredient>();
            this.Steps = new List<Step>();
        }

		public string Title { get; set; }

        public string OriginalUrl { get; set; }
        
        public string Language { get; set; }
        
        public string Note { get; set; }
        
        public string ImageUrl { get; set; }
		
		public string Summary { get; set; }
		
		public string OriginalServings { get; set; }
		
		public IList<Duration> Durations { get; set; }

        public IList<Subrecipe> Subrecipes { get; set; }

        public IList<Ingredient> Ingredients { get; set; }

        public IList<Step> Steps { get; set; }
	}
}